src/clj/fluree/db/json_ld/transact.cljc

```diff
@@ -184,12 +184,12 @@
 (defn final-db
   "Returns map of all elements for a stage transaction required to create an
   updated db."
-  [db new-flakes {:keys [stage-update? policy t txn-id author-did] :as _tx-state}]
+  [db new-flakes {:keys [stage-update? policy t txn author-did] :as _tx-state}]
   (let [[add remove] (if stage-update?
                        (stage-update-novelty (get-in db [:novelty :spot]) new-flakes)
                        [new-flakes nil])
         db-after  (-> db
-                      (update :txns (fnil conj []) [txn-id author-did])
+                      (update :staged conj [txn author-did])
```

Comment:
Are we good with generating lists (instead of vectors) on nil here?

Reply:
I don't think so, lists conj to the front and vectors conj to the back, so the order would be reversed.

Reply:
Yeah, I think this is fine. This should never be `nil` because it's [set in the db `create` function](https://github.com/fluree/db/blob/b6d41fb528c93573e9ce08d2f98549ef7b75ba90/src/clj/fluree/db/db/json_ld.cljc#L186), and it's also an explicit field on the db record.